addons/web/static/tests/search/search_bar_tests.js

```diff
@@ -1718,4 +1719,45 @@ QUnit.module("Search", (hooks) => {
         assert.deepEqual(getFacetTexts(target), ["Bar is in ( First record )"]);
         assert.verifySteps([`/web/domain/validate`]);
     });
+
+    QUnit.test("select autocompleted many2one with allowed_company_ids domain", async function (assert) {
+        assert.expect(5);
+
+        patchUserContextWithCleanup({
+            allowed_company_ids: [1, 5],
+        });
+        
+        await makeWithSearch({
+            serverData,
+            resModel: "partner",
+            Component: SearchBar,
+            searchMenuTypes: [],
+            searchViewId: false,
+            searchViewArch: `
+                    <search>
+                        <field name="foo"/>
+                        <field name="bar" domain="[('company', 'in', allowed_company_ids)]"/>
+                    </search>
+                `,
+        });
+
+        await editSearch(target, "rec");
+        await click(target.querySelector(".o_expand"));
```

Comment:
if there is only one el with o_expand, you can use the shorter form click(target, ".o_expand")